src/integration/errorHandling.test.ts

```diff
@@ -0,0 +1,82 @@
+import { PineconeConnectionError } from '../errors';
+import { Pinecone } from '../index';
+
+describe('Error handling', () => {
+  describe('when environment is wrong', () => {
+    test('calling control plane', async () => {
+      const p = new Pinecone({
+        apiKey: process.env.PINECONE_API_KEY || '',
+        environment: 'wrong-environment',
+      });
+
+      try {
+        await p.listIndexes();
+      } catch (e) {
+        const err = e as PineconeConnectionError;
+        expect(err.name).toEqual('PineconeConnectionError');
+        expect(err.message).toEqual(
+          'Request failed to reach Pinecone. This can occur for reasons such as incorrect configuration (environment, project id, index name), network problems that prevent the request from being completed, or a Pinecone API outage. Check your client configuration, check your network connection, and visit https://status.pinecone.io/ to see whether any outages are ongoing.'
+        );
+        expect(err.cause).toBeDefined();
+      }
+    });
+
+    test('calling data plane', async () => {
+      const p = new Pinecone({
+        apiKey: 'api-key-2',
```

Comment:
**question:** Should this still be `process.env.PINECONE_API_KEY`? Asking as the section is described as `when environment is wrong`.

Reply:
I can change it to make it more clear that the environment is the only thing that matters here. Ultimately only the `environment` value matters for this test since it's an input into the url and there is no `wrong-environment` environment in production.